packages/react-reconciler/src/ReactFiberCompleteWork.js

```diff
@@ -859,6 +859,10 @@ function completeWork(
             if ((workInProgress.effectTag & DidCapture) === NoEffect) {
               // This boundary did not suspend so it's now hydrated and unsuspended.
               workInProgress.memoizedState = null;
+              if (enableSuspenseCallback) {
+                // Notify the callback.
+                workInProgress.effectTag |= Update;
```

Comment:
Seems a bit worrying to have a generic effect tag like this behind a flag since this may significantly fork execution path wherever we check for `Update`.

Reply:
Yea, that's a fair concern. I'm not very confident that this will be a lasting API form though so wouldn't want to do a big refactor yet nor should be pay the cost if it's not needed.

The event replaying might need this flag too and that might move out to a shared path sooner.